hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -87,11 +88,15 @@ public TypedTokenStore(
 			AccountStore accountStore,
 			TransactionRecordService transactionRecordService,
 			Supplier<FCMap<MerkleEntityId, MerkleToken>> tokens,
-			Supplier<FCInvertibleHashMap<MerkleUniqueTokenId, MerkleUniqueToken, OwnerIdentifier>> uniqueTokens,
+			Supplier<FCMap<MerkleUniqueTokenId, MerkleUniqueToken>> uniqueTokens,
+			Supplier<FCOneToManyRelation<EntityId, MerkleUniqueTokenId>> uniqueOwnership,
+			Supplier<FCOneToManyRelation<EntityId, MerkleUniqueTokenId>> uniqueTokenAssociations,
 			Supplier<FCMap<MerkleEntityAssociation, MerkleTokenRelStatus>> tokenRels,
 			BackingTokenRels backingTokenRels
 	) {
```

Comment:
In `persistToken` for minting you should not only do `uniqueTokens.get().put(merkleUniqueId, merkleUniqueToken);` but you should:
1) Add new `uniqueTokenAssociation` between the `Token` and the `newly created MerkleUniqueTokenId`
2) Add new `uniqueOwnership` between the `Treasury` and the `newly created MerkleUniqueTokenId`

Reply:
Don't we do this in the `persistTrackers` method? It is a generalized logic for all operations, when persisting `Change` objects to the new data structures.